politeiad/plugins/pi/pi.go

```diff
@@ -216,3 +242,59 @@ type ProposalMetadata struct {
 	EndDate   int64  `json:"enddate"`   // Estimated end date, Unix time
 	Domain    string `json:"domain"`    // Proposal domain
 }
+
+// BillingStatusT represents the billing status of a proposal that has been
+// approved by the Decrd stakeholders.
+type BillingStatusT uint32
+
+const (
+	// BillingStatusInvalid is an invalid billing status.
+	BillingStatusInvalid BillingStatusT = 0
+
+	// BillingStatusClosed represents a proposal that was approved by
+	// the Decred stakeholders, but has been closed by an admin prior
+	// to the proposal being completed. The most common reason for this
+	// is because a proposal author failed to deliver on the work that
+	// was funded in the proposal. A closed proposal can no longer be
+	// billed against.
+	BillingStatusClosed BillingStatusT = 1
+
+	// BillingStatusCompleted represents a proposal that was approved
+	// by the Decred stakeholders and has been successfully completed.
+	// A completed proposal can no longer be billed against. A proposal
+	// is marked as completed by an admin.
+	BillingStatusCompleted BillingStatusT = 2
+)
+
+// BillingStatusChange represents the structure that is saved to disk when
+// a proposal has its billing status updated. Some billing status changes
+// require a reason to be given.
+//
+// Signature is the admin signature of the Token+Status+Reason.
+type BillingStatusChange struct {
```

Comment:
We should probably start being more specific with our documentation.  Something like this.

```
// BillingStatusChange represents the structure that is saved to disk when
// a proposal has its billing status updated. Some billing status changes
// require a reason to be given. Only admins can update the billing status
// of a proposal.
//
// PublicKey is the admin public key that can be used to verify the signature.
//
// Signature is the admin signature of the Token+Status+Reason.
//
// Receipt is the server signature of the admin signature.
//
// The PublicKey, Signature, and Receipt are all hex encoded and use the
// ed25519 signature scheme.
type BillingStatusChange struct {
	Token     string         `json:"token"`
	Status    BillingStatusT `json:"status"`
	Reason    string         `json:"reason,omitempty"`
	PublicKey string         `json:"publickey"`
	Signature string         `json:"signature"`
	Receipt   string         `json:"receipt"`
	Timestamp int64          `json:"timestamp"` // Unix timestamp
}

// SetBillingStatus sets the billing status of a proposal. Some billing status
// changes require a reason to be given. Only admins can update the billing
// status of a proposal.
//
// PublicKey is the admin public key that can be used to verify the signature.
//
// Signature is the admin signature of the Token+Status+Reason.
//
// The PublicKey and Signature are hex encoded and use the ed25519 signature
// scheme.
type SetBillingStatus struct {
	Token     string         `json:"token"`
	Status    BillingStatusT `json:"status"`
	Reason    string         `json:"reason,omitempty"`
	PublicKey string         `json:"publickey"`
	Signature string         `json:"signature"`
}

// SetBillingStatusReply is the reply to the SetBillingStatus command.
//
// Receipt is the server signature of the client signature. It is hex encoded
// and uses the ed25519 signature scheme.
type SetBillingStatusReply struct {
	Receipt   string `json:"receipt"`
	Timestamp int64  `json:"timestamp"` // Unix timestamp
}
```